modules/crud/schema.go

```diff
@@ -132,7 +137,116 @@ func getFieldType(fieldType ast.Type, fieldTypeStuct *SchemaFieldType, doc *ast.
 	return nil
 }
 
+func (m *Module) schemaValidator(dbType, col string, fields map[string]interface{}) error {
+
+	collectionFields := m.schema[dbType][col]
+	for fieldKey, fieldValue := range collectionFields {
+		// TODO: check if internal elements of list is required for this we need to change schmea parser
+		// check if key is required
+		value, ok := fields[fieldKey]
+		if fieldValue.IsFieldTypeRequired {
+			if !ok {
+				return errors.New("Field " + fieldKey + " Not Present")
+			}
+		}
+
+		// check type
+		fieldType, err := m.checkType(value, fieldValue.Kind, dbType, fieldValue.TableJoin)
+		if err != nil {
+			return err
+		}
+
+		if fieldType != fieldValue.Kind {
+			return errors.New("Wrong Type Wanter " + fieldValue.Kind + " got " + fieldType)
+		}
+
+	}
+
+	return nil
+}
+
 // ValidateSchema checks data type
-func (m *Module) ValidateSchema() {
+func (m *Module) ValidateSchema(dbType, col string, req *model.CreateRequest) error {
+
+	v := make([]map[string]interface{}, 1)
+
+	switch t := req.Document.(type) {
+	case []map[string]interface{}:
+		v = t
+	case map[string]interface{}:
+		v = append(v, t)
+	}
+
+	for _, fields := range v {
+		err := m.schemaValidator(dbType, col, fields)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *Module) checkType(value interface{}, kind, dbType, col string) (string, error) {
+
+	switch v := value.(type) {
+	case int:
+		// TODO: int64
+		switch kind {
+		case TypeDateTime:
+			unitTimeInRFC3339 := time.Unix(int64(v), 0).Format(time.RFC3339)
+			if unitTimeInRFC3339 == "" {
+				return "", errors.New("Integer Wrong Date-Time Format")
+			}
+			return TypeDateTime, nil
+		case TypeID:
+			return TypeID, nil
+		}
+		return TypeInteger, nil
+
+	case string:
+		switch kind {
+		case TypeDateTime:
+			_, err := time.Parse(time.RFC3339, v)
+			if err != nil {
+				return "", errors.New("String Wrong Date-Time Format")
+			}
+			return TypeDateTime, nil
+		case TypeID:
+			return TypeID, nil
+		}
+		return TypeString, nil
+
+	case float32, float64:
+		return TypeFloat, nil
+
+	case bool:
+		return TypeBoolean, nil
+
+	case map[string]interface{}:
+		err := m.schemaValidator(dbType, col, v)
+		if err != nil {
+			return "", err
+		}
+		return TypeJoin, nil
```

Comment:
What if its an embedded document

Reply:
do u mean nested objects

Reply:
it will work for nested objects

Reply:
But you are returning the wrong type in this case. Even if this works, someone reading it might think there is a problem here.